File: cpp/62c_1730A_Planets.cpp
#include<bits/stdc++.h>

using namespace std;

int solve(map<int, int> hmap, int c){
    int cnt = 0;
    for(auto i : hmap){
        cnt += min(c, i.second);
    }
    return cnt;
}

int main(){
    ios::sync_with_stdio(0);
    cin.tie(0);

    int t, n, c, temp;
    vector<int> ans;
    cin>>t;

    for(int i = 0; i < t; i++){
        cin>>n>>c;
        map<int, int> hmap;
        for(int j = 0; j < n; j++){
            cin>>temp;
            hmap[temp]++;
        }
        ans.push_back(solve(hmap, c));
    }

    for(auto i : ans){
        cout<<i<<'\n';
    }
    return 0;
}

File: python/62c_1730A_Planets.py
def solve(arr, n, c):
    hmap = {}
    cnt, i = 0, 0
    arr.sort()
    for i in arr:
        if i in hmap:
            hmap[i] += 1
        else:
            hmap[i] = 1
    for i in hmap:
        cnt += min(hmap[i], c)
    return cnt

def main():
    ans = []
    t = int(input())
    for _ in range(t):
        n, c = map(int, input().split())
        arr = list(map(int, input().split()))
        ans.append(solve(arr, n, c))
    for i in ans:
        print(i)

if __name__ == "__main__":
    main()